common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1175,16 +1178,22 @@ def render(self, block, view_name, context=None):
 
         finally:
             course_id = getattr(self, 'course_id', '')
-            dog_stats_api.increment(XMODULE_METRIC_NAME, tags=[
+            tags=[
                 u'view_name:{}'.format(view_name),
                 u'action:render',
                 u'action_status:{}'.format(status),
                 u'course_id:{}'.format(course_id),
                 u'block_type:{}'.format(block.scope_ids.block_type)
-            ])
+            ]
+            dog_stats_api.increment(XMODULE_METRIC_NAME, tags=tags)
+            dog_stats_api.histogram(
+                XMODULE_DURATION_METRIC_NAME,
+                time.time()-start_time,
```

Comment:
Nit: How much time will the code between `finally` and here take? If it's non-trivial in relation to the `render`\`handle` times, then perhaps capture those times immediately after `finally`.